nilearn/plotting/surf_plotting.py

```diff
@@ -341,9 +341,8 @@ def _get_ticks_matplotlib(vmin, vmax, cbar_tick_format):
     n_ticks = 5
     # ...unless we are dealing with integers with a small range
     # in this case, we reduce the number of ticks
-    if cbar_tick_format == "%i" and vmax - vmin < n_ticks:
+    if cbar_tick_format == "%i" and vmax - vmin < n_ticks - 1:
         ticks = np.arange(vmin, vmax + 1)
-        n_ticks = len(ticks)
     else:
         ticks = np.linspace(vmin, vmax, n_ticks)
```

Comment:
What if `vmin == vmax`? Shouldn't we keep just one single tick, like in the `"%i"` case?

Reply:
More generally, what if the difference between `vmin` and `vmax` is small enough?
```python
>>> vmin = 100
>>> vmax = np.nextafter(vmin, vmin + 1)
>>> n_ticks = 5
>>> 
>>> np.linspace(vmin, vmax, n_ticks)
array([100., 100., 100., 100., 100.])
>>> 
```
```python
>>> vmin = 0
>>> vmax = np.nextafter(vmin, vmin + 1)
>>> n_ticks = 5
>>> 
>>> np.linspace(vmin, vmax, n_ticks)
array([0.e+000, 0.e+000, 0.e+000, 5.e-324, 5.e-324])
>>> 
```
Of course, it's a corner case. Besides, it would only result in multiple ticks drawn at the exact same place. Do we worry about it?

Reply:
It makes sense to keep a single tick when `vmin == vmax`. When calling `_get_ticks_matplotlib(0.1 + 0.2, 0.3, "%f")`, however, note that `vmin == vmax` is `False` so you'll have to do something like `cbar_tick_format % vmin == cbar_tick_format % vmax`:
```
>>> fmt = '%f'
>>> vmin = 0.1 + 0.2
>>> vmax = 0.3
>>> vmin == vmax
False
>>> fmt % vmin == fmt % vmax
True
```
This might partially address your other concern about the difference being small enough where I would posit that "small enough" depends on `cbox_tick_format`.

Reply:
An alternative would be to remove identical ticks, keep only unique keys:
```python
>>> vmin = 100
>>> vmax = 100
>>> n_ticks = 5
>>> 
>>> np.linspace(vmin, vmax, n_ticks)
array([100., 100., 100., 100., 100.])
>>> 
>>> np.array(list(dict.fromkeys(np.linspace(vmin, vmax, n_ticks))))
array([100.])
>>> 
```
```python
>>> vmin = 0
>>> vmax = np.nextafter(vmin, vmin + 1)
>>> n_ticks = 5
>>> 
>>> np.linspace(vmin, vmax, n_ticks)
array([0.e+000, 0.e+000, 0.e+000, 5.e-324, 5.e-324])
>>> 
>>> np.array(list(dict.fromkeys(np.linspace(vmin, vmax, n_ticks))))
array([0.e+000, 5.e-324])
>>> 
```

Reply:
@DimitriPapadopoulos  sounds reasonable. Thx  to both for the discussion & update !

Reply:
@DimitriPapadopoulos nice! I would suggest using `np.unique` to make it easier on the reader:
```
>>> np.unique(np.linspace(vmin, vmax, n_ticks))
array([0.e+000, 5.e-324])
```
What is especially cool is that it fixes one of the unit tests - I also added a test for your use case in the patch below:
```
diff --git a/nilearn/plotting/surf_plotting.py b/nilearn/plotting/surf_plotting.py
index 4ab29eee..90697cca 100644
--- a/nilearn/plotting/surf_plotting.py
+++ b/nilearn/plotting/surf_plotting.py
@@ -344,7 +344,7 @@ def _get_ticks_matplotlib(vmin, vmax, cbar_tick_format):
     if cbar_tick_format == "%i" and vmax - vmin < n_ticks - 1:
         ticks = np.arange(vmin, vmax + 1)
     else:
-        ticks = np.linspace(vmin, vmax, n_ticks)
+        ticks = np.unique(np.linspace(vmin, vmax, n_ticks))
     return ticks


diff --git a/nilearn/plotting/tests/test_surf_plotting.py b/nilearn/plotting/tests/test_surf_plotting.py
index bb88f0ab..23a68520 100644
--- a/nilearn/plotting/tests/test_surf_plotting.py
+++ b/nilearn/plotting/tests/test_surf_plotting.py
@@ -831,10 +831,11 @@ def test_plot_surf_contours_error():
     (1, 5, "%i", [1, 2, 3, 4, 5]),
     (0, 5, "%i", [0, 1.25, 2.5, 3.75, 5]),
     (0, 10, "%i", [0, 2.5, 5, 7.5, 10]),
-    (0, 0, "%.1f", [0, 0, 0, 0, 0]),
+    (0, 0, "%.1f", [0]),
     (0, 1, "%.1f", [0, 0.25, 0.5, 0.75, 1]),
     (1, 2, "%.1f", [1, 1.25, 1.5, 1.75, 2]),
     (1.1, 1.2, "%.1f", [1.1, 1.125, 1.15, 1.175, 1.2]),
+    (0, np.nextafter(0, 1), "%.1f", [0.e+000, 5.e-324]),
 ])
 def test_get_ticks_matplotlib(vmin, vmax, cbar_tick_format, expected):
     ticks = _get_ticks_matplotlib(vmin, vmax, cbar_tick_format)
```

Reply:
I don't know how much portable are the results of the last test from platform to platform, but let's give it a try.

And thanks for `np.unique`, it is self-explaining unlike my unreadable code.